platforms/ios/src/TangramMap/TGMapViewController.h

```diff
@@ -522,36 +524,21 @@ NS_ASSUME_NONNULL_BEGIN
 
  @param path the scene path URL
  @param sceneUpdates a list of `TGSceneUpdate` to apply to the scene
- */
-- (void)loadSceneFileAsync:(NSString *)path sceneUpdates:(NSArray<TGSceneUpdate *> *)sceneUpdates;
-
-/**
- Queue a scene update.
 
- The path is a series of yaml keys separated by a '.' and the value is a string
- of yaml to replace the current value at the given path in the scene.
-
- @param componentPath the path of the scene component to update
- @value the value to assign to the YAML component selected with `componentPath`
-
- @note Scene updates must be applied with `-applySceneUpdates:`
+ @return the integer (SceneID) associated with this scene load or -1 if scene can not be loaded.
  */
-- (void)queueSceneUpdate:(NSString*)componentPath withValue:(NSString*)value;
+- (int)loadSceneFileAsync:(NSString *)path sceneUpdates:(NSArray<TGSceneUpdate *> *)sceneUpdates;
 
 /**
- Queue a list of scene updates.
-
+ Update the scene with the list of updates asyncronously, may call `-[TGMapViewDelegate didLoadScene:withError:]`
+ if a `TGMapViewDelegate` is set to the map view. Also see `TGSceneUpdate` for more info.
+ 
  @param sceneUpdates a list of updates to apply to the scene, see `TGSceneUpdate` for more infos
-
- @note Scene updates must be applied with `-applySceneUpdates:`
- */
-- (void)queueSceneUpdates:(NSArray<TGSceneUpdate *> *)sceneUpdates;
-
-/**
- Apply scene updates queued with `-queueSceneUpdate*` methods.
  If a scene update error happens, scene udpates won't be applied.
+
+ @return the integer (SceneID) associated with this scene update or -1 if scene can not be updated.
  */
-- (void)applySceneUpdates;
+- (int)updateScene:(NSArray<TGSceneUpdate *> *)sceneUpdates;
```

Comment:
Naming is hard!

I am not sure, accordingly to me in both the context it will always be async. I would be good keeping both the platform API consistent. Will update both `ios` and `android` to `updateSceneAsync`. Will also document that these are async tasks, I do not see it documented in doc comments for both platforms.